File: Data Structures & Algorithms/search-2d-matrix/submission-6.py
class Solution:
    def searchMatrix(self, matrix: List[List[int]], target: int) -> bool:
        left = 0
        right = len(matrix) - 1
        while left <= right:
            m = (left + right) // 2
            if target in set(matrix[m]):
                return True
            elif target < matrix[m][-1]:
                right = m - 1
            else:
                left = m + 1 
        return False

File: Data Structures & Algorithms/two-integer-sum/submission-6.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int left = 0;
        int right = 1;
        while (left < nums.length - 1){
            if (nums[left] + nums[right] == target) break;
            right += 1;
            if (right > nums.length - 1) {
                left += 1;
                right = left + 1;
            }
        }
        return new int[]{left, right};
    }
}
